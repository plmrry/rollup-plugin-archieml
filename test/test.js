import vm from "vm";
import { dirname } from "path";
import { fileURLToPath } from "url";
import { rollup } from "rollup";
import archieml from "../dist/index.es.js";

const __dirname = dirname(fileURLToPath(import.meta.url));

test("converts an aml file", async () => {
  const bundle = await rollup({
    input: `${__dirname}/fixtures/basic/index.js`,
    plugins: [archieml()],
  });
  const {
    output: [{ code }],
  } = await bundle.generate({
    format: "esm",
    exports: "default",
  });
  const mod = new vm.SourceTextModule(code);
  await mod.link(() => {});
  await mod.evaluate();
  expect(mod.namespace.default).toMatchSnapshot();
});
