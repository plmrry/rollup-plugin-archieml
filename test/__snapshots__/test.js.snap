// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts an aml file 1`] = `
Object {
  "events": Array [
    Object {
      "type": "header",
      "value": "My Birthday",
    },
    Object {
      "type": "date",
      "value": "August 20th, 1990",
    },
    Object {
      "type": "text",
      "value": "Here's a nice photo",
    },
    Object {
      "type": "image",
      "value": Object {
        "alt": "Family Photo",
        "members": Array [
          "Dad",
          "Mom",
          "Me",
        ],
        "src": "http://example.com/photo.png",
      },
    },
  ],
}
`;
